ui/src/main/webapp/webpack.config.js

```diff
@@ -1,8 +1,8 @@
 module.exports = function(environment) {
   var mode;
 
-  if (environment && environment.hasOwnProperty('environment')) {
```

Comment:
`environment` is object which hold many properties. It is not just a single string value.
Calling the property `environment` again was probably not very clever :) we might call it `mode`? or something else...

It is set using `--env.variable=value` parameter while executing webpack